langchain/src/callbacks/manager.ts

```diff
@@ -348,9 +348,7 @@ export class CallbackManager
     _parentRunId: string | undefined = undefined,
     extraParams: Record<string, unknown> | undefined = undefined
   ): Promise<CallbackManagerForLLMRun[]> {
-    const managers: CallbackManagerForLLMRun[] = [];
-
-    await Promise.all(
+    return Promise.all(
```

Comment:
That's tricky. So the callback in `Promise.map()` is executed whenever that promise is actually finished but then the result of it all is still ordered?

Reply:
the promises in promise.all are all executed simultaneously, all starting at the same time. But promise.all does guarantee that no matter what the order in which they finish the returned array will hold the return values of each in the order they were passed in.